falcon/util/sync.py

```diff
@@ -214,17 +236,13 @@ def async_to_sync(
     Keyword Args:
         **kwargs: Additional args are passed through to the coroutine function.
     """
-
-    # TODO(vytas): The canonical way of doing this for simple use cases is
-    #   asyncio.run(), but that would be a breaking change wrt the above
-    #   documented behaviour; breaking enough to break some of our own tests.
-
-    # NOTE(vytas): Work around get_event_loop deprecation in 3.10 by going via
-    #   get_event_loop_policy(). This should be equivalent for async_to_sync's
-    #   use case as it is currently impossible to invoke run_until_complete()
-    #   from a running loop anyway.
-    loop = asyncio.get_event_loop_policy().get_event_loop()
-    return loop.run_until_complete(coroutine(*args, **kwargs))
+    global _runner
+    # NOTE(vytas): Sometimes our runner's loop can get picked and consumed by
+    #   other utilities and test methods. If that happens, recreate the runner.
+    if _runner.get_loop().is_closed():
```

Comment:
would it make sense to hide this away in a method? Something like

> return _active_runner().run(coroutine(*args, **kwargs))

(that could even be a class instance with a call method

Reply:
What would be the benefits of this change, avoiding `global`?
I can change, for me it works either way.

Reply:
mainly to split the logic of getting an active runner from the execution of the coroutine.
regarding use of global or not, should not make any difference either way so a function is probably easier to do

Reply:
Refactored, check if it looks better now.